edb/schema/schema.py

```diff
@@ -1331,6 +1331,10 @@ def _get_by_id(
             # Avoid the overhead of cast(Object_T) below
             return obj  # type: ignore
 
+    # Important micro-optimization
+    if not TYPE_CHECKING:
+        get_by_id = _get_by_id
```

Comment:
This seems very hacky. Would overloaded functions perform any better here?

Reply:
What's going on here is that `get_by_id` is defined overloaded in the parent, to handle all the type stuff, but then it calls `_get_by_id` which is defined in the child types. The point was to avoid duplicating the big list of overloads for all three types. But it turns out that `get_by_id` was hot enough that eliminating the extra call was noticiable.